src/scripts/main.js

```diff
@@ -1,3 +1,394 @@
 'use strict';
 
-// write your code here
+const table = [
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+];
+
+function up(paramArr, index) {
+  const arr = paramArr.filter(item => item !== 0);
+
+  for (let i = 0; i < arr.length - 1; i++) {
+    const current = arr[i];
+    const next = arr[i + 1];
+
+    if (current === next) {
+      arr[i] = current * 2;
+      scoreValue += arr[i];
+      updateScore();
+      arr[i + 1] = 0;
+    }
+  }
+
+  const mergedArr = arr.filter(item => item !== 0);
+
+  table[0][index] = mergedArr[0] || 0;
+  table[1][index] = mergedArr[1] || 0;
+  table[2][index] = mergedArr[2] || 0;
+  table[3][index] = mergedArr[3] || 0;
+}
+
+function down(paramArr, index) {
+  const arr = paramArr.filter(item => item !== 0);
+
+  for (let i = arr.length - 1; i >= 0; i--) {
+    const current = arr[i];
+    const prev = arr[i - 1];
+
+    if (current === prev) {
+      arr[i] = current * 2;
+      scoreValue += arr[i];
+      updateScore();
+      arr[i - 1] = 0;
+    }
+  }
+
+  const mergedArr = arr.filter(item => item !== 0);
+
+  while (mergedArr.length < 4) {
+    mergedArr.unshift(0);
+  }
+
+  table[0][index] = mergedArr[0] || 0;
+  table[1][index] = mergedArr[1] || 0;
+  table[2][index] = mergedArr[2] || 0;
+  table[3][index] = mergedArr[3] || 0;
```

Comment:
for (let i = 0; i < table.length; i++) {
      table[i][index] = mergedArr[i] || 0;
    }